src/proxy/http/upgrade.rs

```diff
@@ -157,47 +141,39 @@ impl Drop for Inner {
                     .join(client_upgrade)
                     .and_then(|(server_conn, client_conn)| {
                         trace!("HTTP upgrade successful");
-                        tcp::Duplex::new(server_conn, client_conn)
+                        Duplex::new(server_conn, client_conn)
                             .map_err(|e| info!("tcp duplex error: {}", e))
                     });
 
             // There's nothing to do when drain is signaled, we just have to hope
             // the sockets finish soon. However, the drain signal still needs to
             // 'watch' the TCP future so that the process doesn't close early.
-            let fut = self
-                .upgrade_drain_signal
-                .take()
-                .expect("only taken in drop")
-                .watch(both_upgrades, |_| ());
-
-            if let Err(_) = self.upgrade_executor.execute(fut) {
-                trace!("error spawning HTTP upgrade task");
-            }
+            tokio::spawn(
+                self.upgrade_drain_signal
+                    .take()
+                    .expect("only taken in drop")
+                    .watch(both_upgrades, |_| ())
+                    .instrument(Span::current()),
```

Comment:
Nit: if we update to `tracing-futures` 0.1, this can be 
```suggestion
                    .in_current_span(),
```